var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -179,6 +184,8 @@ class Dealii(CMakePackage, CudaPackage):
     # TODO: next line fixes concretization with petsc
     depends_on("hdf5+mpi+hl+fortran", when="+hdf5+mpi+petsc")
     depends_on("hdf5+mpi+hl", when="+hdf5+mpi~petsc")
+    depends_on("kokkos@3.7:", when="@9.5:+kokkos~trilinos")
+    depends_on("kokkos@3.7:+cuda+wrapper", when="@9.5:+kokkos~trilinos+cuda")
```

Comment:
We should probably do
```
    depends_on("kokkos+cuda_lambda", when="~trilinos+cuda")
```
as a workaround (see https://github.com/spack/spack/blob/3f4012cf441bcec98134332338d96bc2642015b1/var/spack/repos/builtin/packages/arborx/package.py#L82C1-L82C60) until we have https://github.com/spack/spack/pull/38957 or https://github.com/spack/spack/pull/38565 and lambda support is enabled by default.

Reply:
Ok. I have no way to test this, so for me it is always a shot in the dark. I'll add your suggestions.